### ISSUE TITLE
Boundary fill  (opengl on and off)

### ISSUE THREAD

**Issue by rgleason** [jongough/ocpn_draw_pi#16]
Most recent compile.
Opengl on - Boundary fill and boundary lines show when zoomed out max only. Just points show for all zooms in.

Opengl off - Boundary fill shows at all levels.

**Comment by rgleason** [comment]
I was trying to put "bug" on this one but can't.  Perhaps you are the only one who can do that?

**Comment by jongough** [comment]
Rick,
   I have tried zooming in and out and the boundaries, for me, display in both opengl and normal mode. I have noticed on a couple of builds that the points dont show up in one or both modes. I normally do a full rebuild and it works again. I suspect that it may be due to code not being recompiled after a change that effects it. Can you try a rebuild of both the mainline code (with a new pull) and of this plugin and let me know what happens.

Thanks

**Comment by rgleason** [comment]
Jon, with Opengl off the boundaries and points work very well.  With Opengl on, I can enter boundaries and points, however they disappear and there is no way to deactivate and activate and get them to show again unitl I turn off opengl and then they all show just fine. This could be some missing lines of code which rewrite the objects after a screen redraw, since they do show as you are entering the bounary.  It just disappears after you hit escape.

**Comment by jongough** [comment]
Rick,
   I am not sure what is going on here. On my Win 7 virtualbox instance I do not see this behaviour, on a win 8 box using your compilations I see the isolated points and the boundary fill disappear when in opengl. Very strange. It seems that opengl is very sensitive and does not give much of an indication of what is wrong. I will look into it.

**Comment by jongough** [comment]
Rick,
   I think I fixed this, found a compiler 'optimisation' that stopped the refreshing of the screen under opengl. Can you give the latest version a go and see if your issue has gone>

**Comment by rgleason** [comment]
I think this is fixed.  After running Chart Scan and Update Database, and full scan, (I had downloaded and updated charts and had done this before, but charts were causing some troubles) Then tried ocpn_draw again. I found that under opengl it worked fine, and the charts turning black were the result of being zoomed in too far. I've been using the plugin with Opengl on for awhile now and it feels comfortable. So I think this is resolved. Thank you Jon
